processor/resourcedetectionprocessor/internal/dynatrace/dynatrace.go

```
@@ -0,0 +1,101 @@
+import (
+	"bufio"
+	"context"
+	"os"
+	"path/filepath"
+	"runtime"
+	"slices"
+	"strings"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/processor"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal"
+)
+
+const TypeStr = "dynatrace"
+
+const dtHostMetadataProperties = "dt_host_metadata.properties"
+
+var dtHostProperties = []string{"dt.entity.host", "host.name"}
+
+type Detector struct {
+	enrichmentDirectory string
+}
+
+func NewDetector(_ processor.Settings, _ internal.DetectorConfig) (internal.Detector, error) {
+	enrichmentDir := "/var/lib/dynatrace/enrichment"
+	if runtime.GOOS == "windows" {
+		// Windows default is "%ProgramData%\dynatrace\enrichment"
+		// If the ProgramData environment variable is not set,
+		// it falls back to C:\ProgramData
+		programDataDir := os.Getenv("ProgramData")
+		if programDataDir == "" {
+			programDataDir = `C:\ProgramData`
+		}
+
+		enrichmentDir = filepath.Join(programDataDir, "dynatrace", "enrichment")
+	}
+
+	return &Detector{
+		enrichmentDirectory: enrichmentDir,
+	}, nil
+}
+
+func (d Detector) Detect(_ context.Context) (pcommon.Resource, string, error) {
+	res := pcommon.NewResource()
+
+	if err := d.readPropertiesFile(res.Attributes()); err != nil {
+		return res, "", err
+	}
+	return res, "", nil
+}
+
+func (d Detector) readPropertiesFile(attributes pcommon.Map) error {
+	filePath := filepath.Join(d.enrichmentDirectory, dtHostMetadataProperties)
+
+	file, err := os.Open(filePath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+		return err
+	}
+
+	defer func() {
+		_ = file.Close()
+	}()
+
+	scanner := bufio.NewScanner(file)
+
+	for scanner.Scan() {
+		// split by the first "=" character. If there is another "=" afterward, this will be part of the value
+		split := strings.SplitN(scanner.Text(), "=", 2)
+		if len(split) != 2 {
+			continue
```

should we debug log or something here?  I can imagine users being confused if their input is silently ignored

---

good point, i have added a debug log there now